web/src/engine/attachment/src/pageContextAttachment.ts

```diff
@@ -0,0 +1,1130 @@
+        this._MutationAdditionObserved(inputElementAdditions[k]);
+      }
+    }
+
+    for(k = 0; k < inputElementRemovals.length; k++) {
+      const elem = inputElementRemovals[k];
+      if(this.isKMWInput(elem)) {
+        this._MutationRemovalObserved(elem);
+      }
+    }
+
+    /* After all mutations have been handled, we need to recompile our .sortedInputs array, but only
+      * if any have actually occurred.
+      */
+    if(inputElementAdditions.length || inputElementRemovals.length) {
+      if(!this.device.touchable) {
+        this.listInputs();
+      } else if(this.device.touchable) {   // If something was added or removed, chances are it's gonna mess up our touch-based layout scheme, so let's update the touch elements.
+        window.setTimeout(() => {
+          this.listInputs();
+        }, 1);
+      }
+    }
+  };
+
+  /**
+   * Function     _MutationAdditionObserved
+   * Scope        Private
+   * @param       {Element}  Pelem     A page input, textarea, or iframe element.
+   * Description  Used by the MutationObserver event handler to properly setup any elements dynamically added to the document post-initialization.
+   *
+   */
+  _MutationAdditionObserved = (Pelem: HTMLElement) => {
+    if(Pelem instanceof Pelem.ownerDocument.defaultView.HTMLIFrameElement) {
+      //Problem:  the iframe is loaded asynchronously, and we must wait for it to load fully before hooking in.
+
+      const attachFunctor = () => {  // Triggers at the same time as iframe's onload property, after its internal document loads.
+        // Provide a minor delay to allow 'load' event handlers to set the design-mode property.
+        window.setTimeout(() => {
+          this.attachToControl(Pelem);
+        }, 1);
+      };
+
+      Pelem.addEventListener('load', attachFunctor);
+
+      // The following block breaks for design-mode iframes, at least in Chrome; a blank document may exist
+      // before the load of the desired actual document.
+      //
+      // /* If the iframe has somehow already loaded, we can't expect the onload event to be raised.  We ought just
+      // * go ahead and perform our callback's contents.
+      // *
+      // * keymanweb.domManager.attachToControl() is now idempotent, so even if our call 'whiffs', it won't cause long-lasting
+      // * problems.
+      // */
+      // if(Pelem.contentDocument.readyState == 'complete') {
+      //   window.setTimeout(attachFunctor, 1);
+      // }
+    } else {
+      this.attachToControl(Pelem);
+    }
+  }
+
+  // Used by the mutation event handler to properly decouple any elements dynamically removed from the document.
+  _MutationRemovalObserved = (Pelem: HTMLElement) => {
+    this.detachFromControl(Pelem); // Remove all KMW event hooks, styling.
+  }
+
+  // To be called by the object responsible for webpage-integration.
+  initMutationObservers(document: Document, manualAttach: boolean) {
```

Comment:
One good and clear example of the top-level-document-only problem I've been referring to:  our MutationObservers.

If `attachMode` is set to 'auto', our observers will note when any context-relevant element is either added to or removed from _the top-level document_, allowing KMW to automatically attach and detach as necessary.  However, testing has shown that this will _not_ work for any elements inserted or removed from within embedded iframe documents - MutationObservers don't cross the iframe boundary.

It's simple enough to fix - "just activate a matching observer set for the embedded document".